core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -1058,6 +1062,24 @@ private[spark] class SparkSubmit extends Logging {
   /** Throw a SparkException with the given error message. */
   private def error(msg: String): Unit = throw new SparkException(msg)
 
+  /**
+   * Store the exit exception using the SparkDiagnosticsSetter.
+   */
+  private def annotateExitException(
+      args: SparkSubmitArguments,
+      sparkConf: SparkConf,
+      throwable: Throwable): Unit = {
+    // Swallow exceptions when storing diagnostics, this shouldn't fail the application.
+    try {
+      if (!isShell(args.primaryResource) && !isSqlShell(args.mainClass)
+          && !isThriftServer(args.mainClass) && !isConnectServer(args.mainClass)) {
+        SparkSubmitUtils.getSparkDiagnosticsSetters(args.master, sparkConf)
```

Comment:
Please remove `sparkConf` because `getSparkDiagnosticsSetters` doesn't use it actually.

```scala
- SparkSubmitUtils.getSparkDiagnosticsSetters(args.master, sparkConf)
+ SparkSubmitUtils.getSparkDiagnosticsSetters(args.master)
```